modules/security-best-practices/pages/rust-canister-development-security-best-practices.adoc

```diff
@@ -0,0 +1,414 @@
+= Rust Canister Development Security Best Practices
+
+== Immutable Smart Contracts
+
+=== Use a decentralized governance system like SNS to make a canister have a decentralized controller
+
+==== Security Concern
+
+The controller of a canister can change / update the canister whenever they like. If a canister e.g. stores assets such as ICP, this effectively means that the controller can steal these by updating the canister and transfer the cycles to their account.
+
+==== Recommendation
+
+To implement an immutable canister smart contract, you cannot be the controller of the canister. Control must be passed to the service nervous system (SNS) or a decentralized governance system.
```

Comment:
is there not a third option? Blackhole the controller so the canister has no existing controller? would that not make it immutable? Also... this section should note that even immutable smart contracts NEED cycles (they are not permament unlike in other blockchains(, and that they can still receive cycles